waiter/waiter.go

```diff
@@ -0,0 +1,77 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+const (
+	// DefaultPollInterval represents duration between checks. Each DefaultPollInterval waiter make the request
```

Comment:
...default duration between operation checks. Waiters accept any [time.Duration] for the interval, but if omitted (set to 0) this one will be used.

And then support this behavior. The logic of how the value is used better be described in `New*`.

Reply:
Updated